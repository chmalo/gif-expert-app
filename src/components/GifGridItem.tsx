const GifGridItem = ({ title, url }: { title?: string; url?: string }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  )
}

export default GifGridItem
