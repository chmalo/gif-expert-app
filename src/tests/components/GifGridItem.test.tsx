import { describe, expect, test } from "vitest"
import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"

describe("test GifGridItem", () => {
  const title = "test title"
  const url = "http://localhost/algo.jpg"

  test("should GifGridItem correctly", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    expect(wrapper).toMatchSnapshot()
  })
})
