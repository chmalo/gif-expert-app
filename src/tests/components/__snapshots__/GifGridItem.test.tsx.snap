// Vitest Snapshot v1

exports[`test GifGridItem > should GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    test title
  </p>
  <img
    alt="test title"
    src="http://localhost/algo.jpg"
  />
</div>
`;
